src/testRunner/unittests/tsserver/regionDiagnostics.ts

```diff
@@ -0,0 +1,76 @@
+import * as ts from "../../_namespaces/ts";
+import {
+    baselineTsserverLogs,
+    TestSession,
+    TestSessionConstructorOptions,
+} from "../helpers/tsserver";
+import { createServerHost } from "../helpers/virtualFileSystemWithWatch";
+
+describe("unittests:: tsserver:: regionDiagnostics", () => {
+    it("diagnostics for select nodes", () => {
+        const file1 = {
+            path: "/a/b/app.ts",
+            content: `function foo(x: number, y: string): number {
+    return x + y;
+}
+
+
+
+foo(10, 50);`,
+        };
+        const host = createServerHost([file1]);
+        const session = new RegionTestSession(host);
+
+        session.executeCommandSeq<ts.server.protocol.OpenRequest>({
+            command: ts.server.protocol.CommandTypes.Open,
+            arguments: { file: file1.path },
+        });
+
+        verifyGetErrRegionRequest({
+            session,
+            files: [{
+                file: file1.path,
+                ranges: [{ startLine: 6, startOffset: 1, endLine: 7, endOffset: 13 }],
+            }],
+        });
+
+        baselineTsserverLogs("regionDiagnostics", "diagnostics for select nodes", session);
+    });
+});
+
+interface VerifyGetErrRegionRequest {

```

Comment:
Please update `VerifyGetErrRequest` in `src\testRunner\unittests\helpers\tsserver.ts` to handle this instead so we can later in time add repros that contain these type of error requests easily

- Change files property to `(string | File | { file, range} ) []` and `SkipErrors` to contain `region?: true` so that we know when the `region` diagnsotics are present and need to postpone semantic and suggestion check to later. (modify `checkAllErrors`) to handle that.

Reply:
I did something slightly different, which is to detect whether region diagnostics should be skipped or not based on the `ranges` property on the `file`, inside `checkAllErrors`. Let me know if that seems good.